src/mono/wasm/runtime-test.js

```diff
@@ -365,42 +356,37 @@ var App = {
 			}
 			wasm_set_main_args (main_argc, main_argv);
 
+			function isThenable (js_obj) {
+				// When using an external Promise library the Promise.resolve may not be sufficient
+				// to identify the object as a Promise.
+				return Promise.resolve (js_obj) === js_obj ||
+						((typeof js_obj === "object" || typeof js_obj === "function") && typeof js_obj.then === "function")
+			}
+
 			try {
-				var invoke_args = Module._malloc (4);
-				Module.setValue (invoke_args, app_args, "i32");
-				var eh_exc = Module._malloc (4);
-				Module.setValue (eh_exc, 0, "i32");
-				var res = runtime_invoke (main_method, 0, invoke_args, eh_exc);
-				var eh_res = Module.getValue (eh_exc, "i32");
-				if (eh_res != 0) {
-					print ("Exception:" + string_get_utf8 (res));
-					test_exit (1);
+				// Automatic signature isn't working correctly
+				let exit_code = Module.mono_call_assembly_entry_point (main_assembly_name, [app_args], "mi");
```

Comment:
This signature has types for two arguments but you're only passing one argument

Reply:
Fixed 

> > It isn't really a sample, it is a runtime test see #44016
> 
> Makes sense, the 1 second sleep made it look like do-not-commit code

1 ms